spark/src/main/java/org/apache/iceberg/spark/SparkFilters.java

```diff
@@ -122,11 +122,7 @@ public static Expression convert(Filter filter) {
 
         case IN:
           In inFilter = (In) filter;
-          Expression in = alwaysFalse();
-          for (Object value : inFilter.values()) {
-            in = or(in, equal(inFilter.attribute(), convertLiteral(value)));
-          }
-          return in;
+          return in(inFilter.attribute(), inFilter.values());
```

Comment:
This dropped the call to `convertLiteral` for each value. That converts values that Spark uses into Iceberg values. I think we still need to call it:

```java
          In inFilter = (In) filter;
          List<Object> nonNullLiterals = Stream.of(inFilter.values())
              .filter(Objects::nonNull)
              .map(SparkFilters::convertLiteral)
              .collect(Collectors.toList());
          boolean hasNull = Stream.of(inFilter.values()).anyMatch(Objects::isNull);
          Expression in = Expressions.in(inFilter.attribute(), nonNullLiterals);
          if (hasNull) {
            return or(isNull(inFilter.attribute()), in);
          } else {
            return in;
          }
```

Reply:
I'm updating the code above with better null handling.